task3.py

```diff
@@ -0,0 +1,21 @@
+"""
+Задать натуральное число k.
+Сформируйте многочлен (полином) степени k со случайными коэффициентами из промежутка от 0 до 100, включительно.
+Многочлен вывести в консоль и записать в файл.
+Ввод: значение типа <int>
+Вывод: значение типа <str>, файл с одной строкой.
+Пример:
+2
+2x^2 + 4x + 5 = 0
+"""
+
+from random import randint
+max_val=100
+k = int(input('Введите натуральную степень k:'))
+koeff=[randint(0,max_val) for i in range(k)]+[randint(1,max_val)]
+poly='+'.join([f'{(j,"")[j==1]}x^{i}' for i, j in enumerate(koeff) if j][::-1])
+poly=poly.replace('x^1+', 'x+')
+poly=poly.replace('x^0', '')
+poly+=('','1')[poly[-1]=='+']
+poly=(poly, poly[:-2])[poly[-2:]=='^1']
+print(poly)
```

Comment:
Наталья, почему не соблюдаете PEP8?
```suggestion
from random import randint

max_val = 100
k = int(input('Введите натуральную степень k:'))
koeff = [randint(0, max_val) for i in range(k)] + [randint(1, max_val)]
poly = '+'.join([f'{(j, "")[j == 1]}x^{i}' for i, j in enumerate(koeff) if j][::-1])
poly = poly.replace('x^1+', 'x+')
poly = poly.replace('x^0', '')
poly += ('', '1')[poly[-1] == '+']
poly = (poly, poly[:-2])[poly[-2:] == '^1']
print(poly)

```

Reply:
На самом деле, ваше решение достаточно неплохое. Я еще не видел решения с таким небольшим количеством строк кода. Оно работает почти правильно. Исключение составляет только пару замечаний. Например, если все коэффициенты будут равны нулю, у вас произойдет ошибка. Кроме того в записи многочлена отсутствует `= 0`.
Ну и еще одно замечание это то, что у вас не происходит записи в файл.